.github/workflows/ci.yml

```
@@ -9,45 +14,75 @@ defaults:
     shell: bash -l {0}
 
 jobs:
+  get_workflow_tests:
+    runs-on: ubuntu-latest
+    outputs:
+      test_list: ${{ steps.get_tests.outputs.tests }}
+    steps:
+      - uses: actions/checkout@v2.3.4 # https://github.com/actions/checkout.
+        with:  # Use fastest possible checkout.
+          submodules: false
+          fetch-depth: 1
+      - id: get_tests
+        run: |
+          # Remove all whitespace first
+          TESTS="$(cat ci_tests.json | tr -d '\r\n\t ')"
```

```suggestion
          TESTS="$(cat ci_tests.json | jq -c)"
```

might be a safer version of that, in case at some point strings have whitespace in them.

---

Sure. I will commit this and we can see if `jq` is installed by default on the nodes. If so it is indeed much better to use a parser rather than pattern matching.

---

It works. Nice!

---

yay!